core/src/main/scala/io/finch/request/package.scala

```diff
@@ -162,6 +180,8 @@ package object request {
       def apply[Req](req: Req)(implicit ev: Req => HttpRequest) =
         try number.toFuture
         catch { case _: IllegalArgumentException => ValidationFailed(param, rule).toFutureException }
+
+      override def toString: String = s"param: $param"
```

Comment:
There is one thing I just noticed now and might need to be adjusted in my follow-up work: The should method produces exceptions of type `RequestReaderError` which is a base class and not a case class which breaks the promise made in the documentation: "all the exception throw by `RequestReader` are case classes".

What type of Error should the rules throw instead? `ValidationFailed` is already used by the number converters and has two properties, whereas we need a case class with a simple message property.